### PR TITLE
configure ef

### PR DESCRIPTION
**Pull request by Roma-Jivchik**

### CODE REVIEW

File: DistributionSystemApi/DistributionSystemApi.DataAccess/EntitiesBuilder/NotificationTemplateEntityTypeConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+{
+    public class NotificationTemplateEntityTypeConfiguration : IEntityTypeConfiguration<NotificationTemplate>
+    {
+        public void Configure(EntityTypeBuilder<NotificationTemplate> builder)
+        {
+            builder
+                .Property(e => e.Id)
+                .HasMaxLength(200)
+                .IsUnicode(false)
+                .HasColumnName("ID");
```
Comment (EugeneSeledtsov) [inline DistributionSystemApi/DistributionSystemApi.DataAccess/EntitiesBuilder/NotificationTemplateEntityTypeConfiguration.cs]: "Id"

File: DistributionSystemApi/DistributionSystemApi.DataAccess/Interfaces/IDataContext.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace DistributionSystemApi.Data.Interfaces
+{
+    public interface IDataContext
+    {
+        IQueryable<TEntity> Get<TEntity>() where TEntity : BaseEntity;
+
+        void Create<TEntity>(TEntity entity, CancellationToken cancellationToken) where TEntity : BaseEntity;
```
Comment (JenBytowski) [inline DistributionSystemApi/DistributionSystemApi.DataAccess/Interfaces/IDataContext.cs]: return type void and cancellation token?)

File: DistributionSystemApi/DistributionSystemApi.DataAccess/DataContext.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace DistributionSystemApi.Data
+{
+    public class DataContext : IDataContext
+    {
+        private readonly ContentDistributionSystemContext contentDistributionSystemContext;
+
+        public async void Create<TEntity>(TEntity entity, CancellationToken cancellationToken) where TEntity : BaseEntity
+        {
+            entity.Id = Guid.NewGuid();
```
Comment (JenBytowski) [inline DistributionSystemApi/DistributionSystemApi.DataAccess/DataContext.cs]: ef will do it automatically for you

File: DistributionSystemApi/DistributionSystemApi.DataAccess/DataContext.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+            contentDistributionSystemContext.Set<TEntity>().Add(entity);
+
+            await SaveChangesAsync(cancellationToken);
```
Comment (JenBytowski) [inline DistributionSystemApi/DistributionSystemApi.DataAccess/DataContext.cs]: we dont want to save changes there, we only want to wrap existed functionality of dbcontext

File: DistributionSystemApi/DistributionSystemApi.DataAccess/DataContext.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        public IQueryable<TEntity> Get<TEntity>() where TEntity : BaseEntity
+        {
+            return contentDistributionSystemContext.Set<TEntity>();
```
Comment (JenBytowski) [inline DistributionSystemApi/DistributionSystemApi.DataAccess/DataContext.cs:19]: the same as above

File: DistributionSystemApi/DistributionSystemApi/ConfigurationExtensions/DatabaseConfigurationExtension.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿using DistributionSystemApi.Data;
+using Microsoft.EntityFrameworkCore;
+
+namespace DistributionSystemApi.ConfigurationExtensions
+{
+    public static class DatabaseConfigurationExtension
+    {
+        public static void ConfigureServices(this WebApplicationBuilder builder)
```
Comment (JenBytowski) [inline DistributionSystemApi/DistributionSystemApi/ConfigurationExtensions/DatabaseConfigurationExtension.cs]: please use IServiceCollection instead(Check mail service configuration code)

File: DistributionSystemApi/DistributionSystemApi/appsettings.json [hunk 7]
```diff
@@ -5,5 +5,8 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
-  "AllowedHosts": "*"
+  "AllowedHosts": "*",
+  "ConnectionStrings": {
```
Comment (JenBytowski) [inline DistributionSystemApi/DistributionSystemApi/appsettings.json]: please also add this section to appsettings.development.json

File: DistributionSystemApi/DistributionSystemApi/ConfigurationExtensions/DatabaseConfigurationExtension.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
+﻿using DistributionSystemApi.Data;
+using Microsoft.EntityFrameworkCore;
+
+namespace DistributionSystemApi.ConfigurationExtensions
+{
+    public static class DatabaseConfigurationExtension
+    {
+        public static void ConfigureServices(this IServiceCollection services, IConfiguration configuration)
```
Comment (JenBytowski) [inline DistributionSystemApi/DistributionSystemApi/ConfigurationExtensions/DatabaseConfigurationExtension.cs]: those extensions should return IServiceCollection like microsofts configuration extensions do

Comment (JenBytowski) [reply DistributionSystemApi/DistributionSystemApi/ConfigurationExtensions/DatabaseConfigurationExtension.cs]: Also its a bit strange name

**Review by JenBytowski — APPROVED** [review]

**Review by EugeneSeledtsov — APPROVED** [review]

**Pull request merged**